traveler-swift-core/TravelerKit/Models/BookingItemQuery.swift

```diff
@@ -64,7 +86,7 @@ public struct BookingItemQuery {
      */
 
     public func filterSearchWith(_ filters: BookingItemSearchFilters) -> BookingItemQuery {
-        return BookingItemQuery(text: self.text, range: filters.priceRange, categories: filters.categories ?? [] , boundingBox: self.boundingBox)
+        return BookingItemQuery(text: filters.text ?? self.text, range: filters.priceRange ?? self.priceRange, categories: filters.categories ?? self.categories , boundingBox: self.boundingBox, country: filters.country ?? self.country, city: filters.city ?? self.city, sortOption: filters.sortOption ?? self.sortOption, sortOrder: filters.sortOrder ?? self.sortOrder)
     }
 
     func isValid() -> Bool {
```

Comment:
check if validation rule for query remains the same even with the new attributes. The rule is that at least one parameter is set. Now that we have city I would assume the rule still stands and it would need to be reflected here.

Reply:
You're right, added validation for nil city as well